### PR TITLE
Move all code to src/lib.rs instead of resolver.rs for simplicity.

### PR DESCRIPTION
**Pull request by bowrocker**
No need to use a separate file. All tests pass on my local machine.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,18 +6,19 @@ edition = "2018"
 
 [dependencies]
 base64 = "0.10.1"
+chrono = "0.4"
+cueball = { git = "https://github.com/joyent/rust-cueball.git" }
 derive_more = "0.14.0"
+futures = "0.1"
+quickcheck = "0.8.0"
```
Comment (kellymclaughlin) [inline Cargo.toml]: Do you need quickcheck? Wasn't sure if you planned on testing with it or if it's unnecessary.

File: src/lib.rs [hunk 2]
```diff
@@ -4,4 +4,568 @@
 
 //! resolver is a library for resolving DNS records for cueball.
 #[allow(dead_code)] // TODO: Remove after initial dev
-pub mod resolver;
+
+use chrono::prelude::*;
```
Comment (kellymclaughlin) [inline src/lib.rs]: I'm going to use this as an opportunity to solicit your opinion on a convention I've adopted. I've been organizing my imports into three groups. The first is the `std` imports, the second is the non-`std` external imports and the third is any crate-local imports. [Here's](https://github.com/joyent/rust-cueball/blob/master/src/connection_pool.rs) an example. I find it a nice way to bring a bit of order to what can get to be an overwhelmingly long list of jumbled imports, but curious if you see any value in such a practice.

File: src/lib.rs [hunk 3]
```diff
@@ -4,4 +4,568 @@
...
+            log: self.log.clone(),
+        });
+
+        let _fsm = thread::spawn(move || {
+            let _result = resolver.wait();
+        });
+    }
+
+    fn stop(&mut self) {
+        std::unimplemented!()
```
Comment (kellymclaughlin) [inline src/lib.rs:134]: I'm curious what this implementation will look like. Will you be able to initiate an FSM transition to the Stop state from this thread?

File: src/lib.rs (hunk 2 above)
Comment (bowrocker) [reply src/lib.rs]: I like it -- will update.

File: Cargo.toml (hunk 1 above)
Comment (bowrocker) [reply Cargo.toml]: Good catch, will remove (for now).

File: src/lib.rs (hunk 3 above)
Comment (bowrocker) [reply src/lib.rs:134]: That was the idea, though I am fuzzy on the implementation right now. Were you looking for something different?

Comment (kellymclaughlin) [reply src/lib.rs:134]: No, I don't have enough familiarity with the details yet to have an opinion on how it should be done. It was really just curiosity that made me comment after a first look at the changes.

File: src/lib.rs (hunk 2 above)
Comment (kellymclaughlin) [reply src/lib.rs]: I chatted with rm about this briefly on Friday an we might even further break this down and separate third party external imports from the joyent external imports.

**Pull request merged**